### ISSUE TITLE
HTML Compression - Page generation speeds

### ISSUE THREAD

**Issue by sdesign75** [wpsharks/comet-cache#655]
Hi

The Html compression seems to really slow down page generation on my site (see my figures below). 

Obviously for pages that have already been cached the html compression is going to make them load faster (at least I think it will), but what it means is that the first visitor to a page has over twice the page generation time that they had without html compression. Not good!  

What do you think about these figures?

Alex

---
## Home Page

Home page with NO Html/CSS/JS compression: 
- ZenCache file built in 3.39689 seconds

Home page usng "ZenCache" Html/CSS/JS compression: 
- ZenCache HTML Compressor took 3.41688 seconds (overall).
- ZenCache file built in 7.04246 seconds

Home page using only "Auto Optimise" plugin Html/CSS/JS compression (not ZenCache): 
- ZenCache file built in 4.31672 seconds

Home page usng both "ZenCache" and "Auto Optimize" plugin Html/CSS/JS compression together: 
- ZenCache HTML Compressor took 0.58357 seconds (overall).
- ZenCache file built in 5.22767 seconds

---
## Test Post

Test Post using NO Html/CSS/JS compression: 
- ZenCache file built in 3.91072 seconds

Test Post using "ZenCache" Html/CSS/JS compression: 
- ZenCache HTML Compressor took 4.09243 seconds (overall)
- ZenCache file built in 7.09248 seconds

Test Post using "Auto Optimise" plugin Html/CSS/JS compression (not ZenCache): 
- ZenCache file built in 4.98193 seconds

Test Post usng both "ZenCache" and "Auto Optimize" Html/CSS/JS compression: 
- ZenCache HTML Compressor took 0.57843 seconds (overall).
- ZenCache file built in 4.65894 seconds

**Comment by jaswrks** [comment]
@sdesign75 Thanks for posting the stats, that helps.

You wrote...

> Home page with NO Html/CSS/JS compression:
> ZenCache file built in 3.39689 seconds

Can I assume then, that your average page generation time w/o _any_ caching or HTML Compression whatsoever is somewhere in the neighborhood of three full seconds?

If that's the case, then it stands to reason that the HTML Compressor is going to take a longer time on this site. 3 seconds is quite a long time. The HTML Compressor not only optimizes the JS, CSS, and markup, it also compresses and consolidates both local and remote resources.

That last point is important. The HTML Compressor collects both local and remote resources. So if each request to your site can take 1 second or more, things could be really slow with the HTML Compressor (on first load), because the site itself (without any caching) is slow to serve these files to the HTML Compressor also. For instance, imagine the HTML Compressor collecting the data it needs (behind-the-scenes) in order to compress and consolidate each of your JS, CSS, or remote files. If those HTTP requests take longer, it can slow down the whole operation a bit.

Ways to improve this might include:
- Optimize the web server itself to make it faster if at all possible.
- Reduce the number of active plugins that you have running, or look for bottlenecks.

If those are no longer options, and you are down to working with the HTML Compressor and ZenCache only at this point in time, then I'd start considering the Auto-Cache Engine:
- See: **Dashboard → ZenCache → Plugin Options → Auto-Cache Engine**
  
  Running this will prime the cache so that your visitors are not impacted by the first-to-visit, slow-to-load scenario nearly as often, or at all in many cases.

**Comment by sdesign75** [comment]
Hi,

Yes, my page generations are around 3 full seconds. It's specifically the buddypress plugin. It really slows things down. My other Wordpress sites on the same server are much less than 2 seconds. This is why I bought ZenCache, was the only cache system that seemed to work properly with member sites. I have turned pre-cache on so this helps. I really just posted the figures for your info as I found it quite interesting myself. 

Cheers
Alex
